streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java

```diff
@@ -166,20 +166,46 @@ public void shouldForwardOldValuesWhenEnabled() {
         store.setFlushListener(cacheFlushListener, true);
         store.put(bytesKey("1"), bytesValue("a"));
         store.flush();
+        assertEquals("a", cacheFlushListener.forwarded.get("1").newValue);
```

Comment:
The augmented unit tests here and below are for 1) demonstrating the optimization, 2) demonstrating the bug fix as well.